timesketch/frontend-ng/src/components/Visualization/VisualizationEditor.vue

```diff
@@ -0,0 +1,418 @@
+                @updateSplitByTimeline="selectedSplitByTimeline = $event"
+              ></TsAggregationConfig>
+              <TsChartConfig
+                :aggregatorType="selectedAggregator"
+                :chartType="selectedChartType"
+                @updateChartType="selectedChartType = $event"
+                :title="selectedChartTitle"
+                @updateTitle="selectedChartTitle = $event"
+                :height="selectedHeight"
+                @updateHeight="selectedHeight = $event"
+                :width="selectedWidth"
+                @updateWidth="selectedWidth = $event"
+                :xTitle="selectedXTitle"
+                @updateXTitle="selectedXTitle = $event"
+                :showXLabels="selectedShowXLabels"
+                @updateShowXLabels="selectedShowXLabels = $event"
+                :yTitle="selectedYTitle"
+                @updateYTitle="selectedYTitle = $event"
+                :showYLabels="selectedShowYLabels"
+                @updateShowYLabels="selectedShowYLabels = $event"
+                :showDataLabels="selectedShowDataLabels"
+                @updateShowDataLabels="selectedShowDataLabels = $event"
+              ></TsChartConfig>
+            </v-card-text>
+          </v-card>
+        </v-col>
+        <v-col cols="8">
+          <v-card outlined>
+            <v-card-text>
+              <TsChartCard
+                v-if="chartSeries && selectedChartType"
+                :fieldName="selectedField.field"
+                :metricName="selectedMetric"
+                :is-time-series="selectedAggregator ? selectedAggregator.endsWith('date_histogram') : false"
+                :chartSeries="chartSeries" 
+                :chartLabels="chartLabels"
+                :chartTitle="selectedChartTitle"
+                :chartType="selectedChartType"
+                :height="selectedHeight"
+                :width="selectedWidth"
+                :xTitle="selectedXTitle"
+                :showXLabels="selectedShowXLabels"
+                :yTitle="selectedYTitle"
+                :showYLabels="selectedShowYLabels"
+                :showDataLabels="selectedShowDataLabels"
+              ></TsChartCard>
+            </v-card-text>
+          </v-card>
+        </v-col>
+      </v-row>
+    </v-card-text>
+    <v-card-actions>
+      <v-btn color="primary" 
+        @click="loadAggregationData"
+        :disabled="!(
+          selectedField && 
+          selectedAggregator && 
+          selectedChartType
+        )"
+      >
+        Load/refresh data
+      </v-btn>
+      <v-btn 
+        color="primary"
+        :disabled="response == null || !selectedChartTitle"
+        @click="saveVisualization"
+      >
+        Save
+      </v-btn>
+      <v-btn 
+        color="primary" 
+        @click="clear"
+      >
+        Clear
+      </v-btn>
+      <v-btn 
+        color="primary" 
+        @click="clear" 
+        :to="{ name: 'Explore' }"
+      >
+        Cancel
+      </v-btn>
```

Comment:
What is your use-case for the cancel button? Default behaviour with all the other views switching happens via the left-panel. The state of the canvas is stored on purpose (but can be deleted by clicking clear).

Reply:
The cancel follows the same behaviour as the Sigma editor where cancel switches the view back to the explore view.